File: problem703.cpp
#include <vector>
#include <iostream>
using namespace std;
class Solution {
public:
    int search(vector<int>& nums, int target) {
        // Binaray search
        int left = 0;
        int right = nums.size()-1;

        while(left <= right){
            // find the midpoint
            int mid = left + (right - left)/2;
            if(nums[mid] == target){
                return mid;
            }
            else if(nums[mid] > target){
                right = mid -1 ; //  check the left side
            }
            else{
                left = mid + 1; // this is the right side
            }
        }
        return -1; 
    }
};

File: problem703.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        # Binaray seach
        # Divide problem into two
        def divide(number, left, right, target):
            if left > right:
                return -1
            mid = (right - left)//2 + left # find the middle point
            if number[mid] == target:
                return mid
            elif number[mid] > target:
                return divide(number,left, mid-1, target) # check in the left side
            elif number[mid] < target: # check in the right side
                 return divide(number, mid+1, right, target)
        return divide(nums, 0, len(nums) -1, target)